ERCS/erc-7795.md

```diff
@@ -0,0 +1,589 @@
+
+Example:
+
+```json
+[
+  {
+    "version": "1.0",
+    "from": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+    "calls": [
+      {
+        "to": "0x631998e91476da5b870d741192fc5cbc55f5a52e",
+        "value": "0x00",
+        "data": "0x...",
+        "chainId": "0x01",
+      }
+    ],
+    "capabilities": {
+      "erc1155MinAllowance": {
+        "version": "1.0",
+        "chainId": "0x01",
+        "owner": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+        "operator": "0xbc4ca0eda7647a8ab7c2061c2e118a18a936f13d",
+        "token": "0x631998e91476da5b870d741192fc5cbc55f5a52e",
+        "tokenId": "0x10",
+        "minAmount": "0x0f4240"
+      }
+    }
+  }
+]
+```
+
+## Rationale
+
+This ERC extends [EIP-5792](./eip-5792.md) rather than defining new RPC methods because:
+
+1. **Consistency**: Leverages existing capability discovery mechanism
+2. **Composability**: Requirements can be combined with other [EIP-5792](./eip-5792.md) capabilities
+3. **Flexibility**: Wallets can implement only the requirements they support
+4. **Extensibility**: New requirement types can be added as additional capabilities
+
+The decision to split requirements into individual capabilities rather than a single capability type allows:
+
+1. Granular support by wallets
+2. Clear capability discovery
+3. Independent versioning of requirement types
+4. Simpler implementation for basic wallets
+
+## Reference Implementation
+
+This ERC serves as a foundational component for building user experiences that rely on cross-chain actions. It can be leveraged in various ways, depending on the combination of use cases and wallet implementations. Consider the following high-level examples.
+
+Not shown in the examples below are the `wallet_getCallsStatus` and `wallet_showCallsStatus` methods, which are used to query the status of a call bundle and display it to the user. The dApp **MAY** use these methods to provide a better user experience while waiting for the transactions to be fulfilled. More information can be found in the [EIP-5792](./eip-5792.md) specification.
+
+### Marketplace Interaction with Bridge Using an EOA Wallet
+
+A user wants to purchase [ERC-721](./eip-721.md) tokens on **Chain A** but only has funds on **Chain B**. The user employs an Externally Owned Account (EOA) based wallet. The dApp requests an intended transaction using `wallet_sendCalls`, which includes the marketplace transaction alongside the requirement of owning funds on Chain A.
+
+In this scenario, the wallet **MAY** prompt the user to sign the necessary transactions to fulfill the requirements before proceeding with the main transaction. The wallet **SHOULD** compute possible solutions to meet the requirements and **MUST** ensure that these prerequisites are met prior to executing the main transaction.
+
+```mermaid
```

Comment:
Our renderer does not support mermaid. Please include an image (preferably SVG) in your assets directory.